src/diagnostics/RemoteDebugCommands.ts

```diff
@@ -0,0 +1,174 @@
+            this.reportMessage('Starting debugging...');
+
+            // Enable tracing for debug configuration
+            debugConfig.trace = 'verbose';
+            await vscode.debug.startDebugging(undefined, debugConfig);
+
+            const terminateDebugListener: vscode.Disposable = vscode.debug.onDidTerminateDebugSession(async (event: vscode.DebugSession) => {
+                if (event.name === debugConfig.name) {
+                    if (tunnelProxy !== undefined) {
+                        tunnelProxy.dispose();
+                    }
+                    terminateDebugListener.dispose();
+
+                    const confirmDisableMessage: string = 'Leaving the app in debugging mode may cause performance issues. Would you like to disable debugging for this app? The app will be restarted.';
+                    await vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async (p: vscode.Progress<{}>): Promise<void> => {
+                        this._progress = p;
+                        await this.setRemoteDebug(false, confirmDisableMessage);
+                    });
+                }
+            });
+        });
+    }
+
+    public async enableRemoteDebug(): Promise<void> {
+        await this.checkForRemoteDebugSupport();
+
+        const confirmMessage: string = 'The app configuration will be updated to enable remote debugging and restarted. Would you like to continue?';
+        const noopMessage: string = 'The app is already configured for debugging.';
+
+        await vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async (p: vscode.Progress<{}>): Promise<void> => {
+            this._progress = p;
+            await this.setRemoteDebug(true, confirmMessage, noopMessage);
+        });
+    }
+
+    public async disableRemoteDebug(): Promise<void> {
+        await this.checkForRemoteDebugSupport();
+
+        const confirmMessage: string = 'The app configuration will be updated to disable remote debugging and restarted. Would you like to continue?';
+        const noopMessage: string = 'The app is not configured for debugging.';
+
+        await vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async (p: vscode.Progress<{}>): Promise<void> => {
+            this._progress = p;
+            await this.setRemoteDebug(false, confirmMessage, noopMessage);
+        });
+    }
+
+    private reportMessage(msg: string): void {
+        this._outputChannel.appendLine(msg);
+
+        if (this._progress) {
+            this._progress.report({ message: msg });
+        }
+    }
+
+    private async getSiteConfig(): Promise<SiteConfigResource> {
+        if (!this._siteConfig) {
+            this.reportMessage('Fetching site configuration...');
+            this._siteConfig = await this._siteClient.getSiteConfig();
+        }
+
+        return this._siteConfig;
+    }
+
+    private async needUpdateSiteConfig(isRemoteDebuggingToBeEnabled: boolean): Promise<boolean> {
```

Comment:
nit: IMO it would be more readable if these two lines of code weren't in their own method (which is only used once)